src/response_manager.zig

```diff
@@ -0,0 +1,280 @@
+            const now = std.time.nanoTimestamp();
+            if (now >= deadline) {
+                self.timeout_expired = true;
+                break;
+            }
+            
+            const remaining_ns = @as(u64, @intCast(deadline - now));
+            self.condition.timedWait(&self.mutex, remaining_ns) catch {
+                self.timeout_expired = true;
+                break;
+            };
+        }
+        
+        // Check if an error occurred
+        if (self.error_status) |err| {
+            return err;
+        }
+        
+        return self.message;
+    }
+    
+    pub fn complete(self: *ResponseInfo, msg: *Message) void {
+        self.mutex.lock();
+        defer self.mutex.unlock();
+        
+        if (!self.completed) {
+            self.message = msg;
+            self.completed = true;
+            self.condition.signal();
+        }
+    }
+    
+    pub fn completeWithError(self: *ResponseInfo, err: anyerror) void {
+        self.mutex.lock();
+        defer self.mutex.unlock();
+        
+        if (!self.completed) {
+            self.error_status = err;
+            self.completed = true;
+            self.condition.signal();
+        }
+    }
+};
+
+pub const ResponseManager = struct {
+    allocator: Allocator,
+    
+    // Response handling state
+    resp_sub_prefix: ?[]u8 = null,
+    resp_sub_len: usize = 0,
+    resp_mux: ?*Subscription = null,
+    resp_map: std.HashMap([]const u8, *ResponseInfo, std.hash_map.StringContext, std.hash_map.default_max_load_percentage),
+    
+    // Token generation state (simple counter)
+    token_counter: u64 = 0,
+    
+    pub fn init(allocator: Allocator) ResponseManager {
+        return ResponseManager{
+            .allocator = allocator,
+            .resp_map = std.HashMap([]const u8, *ResponseInfo, std.hash_map.StringContext, std.hash_map.default_max_load_percentage).init(allocator),
+        };
+    }
+    
+    pub fn deinit(self: *ResponseManager) void {
+        if (self.resp_sub_prefix) |prefix| {
+            self.allocator.free(prefix);
+        }
+        
+        if (self.resp_mux) |sub| {
+            sub.deinit();
+        }
+        
+        // Clean up any remaining response infos
+        var iterator = self.resp_map.iterator();
+        while (iterator.next()) |entry| {
+            self.allocator.free(entry.key_ptr.*);
+            self.allocator.destroy(entry.value_ptr.*);
+        }
+        self.resp_map.deinit();
+    }
+    
+    pub fn ensureInitialized(self: *ResponseManager, connection: anytype) !void {
+        if (self.resp_mux != null) return; // Already initialized
+        
+        // Generate unique inbox prefix
+        const base_inbox = try inbox.newInbox(self.allocator);
+        defer self.allocator.free(base_inbox);
+        
+        self.resp_sub_prefix = try std.fmt.allocPrint(self.allocator, "{s}.", .{base_inbox});
```

Comment:
Since inbox prefix is fixed length, we could just use a statically sized array and change the API to work like this:

inbox.newInbox(buffer[0..X])
buffer[x] = '.'

no need for all these allocations